### ISSUE TITLE
Python 2.7 and lower support

### ISSUE THREAD

**Issue by hetoan2** [robrua/cassiopeia#17]
I installed via "pip install cassiopeia" and I have a python 2.7 installation. I'm currently going through it now and changing the imports from Python 3 to python 2.7, because they look to be not that many. I just think you should have a disclaimer or something.

**Comment by hetoan2** [comment]
I was able to get it working with the [future](http://python-future.org/quickstart.html) and `pip install future`.  Then by adding the lines 
```
from future import standard_library
standard_library.install_aliases()
```

I also did `pip install enum34` for enums, but this might not be required.

**Comment by jjmaldonis** [comment]
There have been a couple people using python 2.7, is there any specific reason you are using it rather than python 3?

**Comment by hetoan2** [comment]
It's mostly just "reasons" so to speak, but I just have a lot of code pre-written in 2.7 at this point.

**Comment by hetoan2** [comment]
I also think the server I'm writing it on is Django compiled for Python 2.7, so I'd really not port a lot of code and whatnot.

Also when I actually ran it, it ended up not working with the fix I stated.

```
>>> summoner = riotapi.get_summoner_by_name("musouka")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/cassiopeia/core/summonerapi.py", line 94, in get_summoner_by_name
    summoner = cassiopeia.dto.summonerapi.get_summoners_by_name(name)[__standardize(name)]
  File "/usr/local/lib/python2.7/dist-packages/cassiopeia/dto/summonerapi.py", line 24, in get_summoners_by_name
    response = cassiopeia.dto.requests.get(request)
  File "/usr/local/lib/python2.7/dist-packages/cassiopeia/dto/requests.py", line 65, in get
    content = rate_limiter.call(executeRequest, url) if rate_limiter else executeRequest(url)
  File "/usr/local/lib/python2.7/dist-packages/cassiopeia/type/api/rates.py", line 109, in call
    return method(*args)
  File "/usr/local/lib/python2.7/dist-packages/cassiopeia/dto/requests.py", line 91, in executeRequest
    response = urllib.request.urlopen(url)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 171, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 494, in open
    response = self._open(req, data)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 517, in _open
    'unknown_open', req)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 466, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 1353, in unknown_open
    raise URLError('unknown url type: %s' % type)
future.backports.urllib.error.URLError: <urlopen error unknown url type: https>
```

There's an error log if you're considering making a branch for Python 2.7; however from your tone I don't think it's likely.

**Comment by jjmaldonis** [comment]
We can see what Rob will say. I also have a lot of code written in 2.7 for another project, but I really should convert. Python 3 was released in 2008, so it's been out for 7-8 years now. Django started supporting Python 3 in version 1.5 and they are on 1.8 now. It's sort of a question of whether it's worth putting in a reasonable amount of effort to make cassio backwards compatible to 7-8 years ago. As you have noticed, it probably isn't simple.  It is an issue of demand, so it's good that you brought this up. Also, if we do support Python 2.7, we have to make sure that all of our future commits (and acceptance of pull requests) work for both 2.7 and 3, which puts another layer of complication on top of a library whose goal is to provide simplicity. That's sort of my personal opinion right now, but I am open to being persuaded otherwise.

**Comment by robrua** [comment]
So to preface this it's worth noting that the vast majority of development of API wrappers I've done is directly driven by a project I'm working on. Cass is, first and foremost, designed to be "the library *I* want to use", and since I don't use 2.x, I don't have a ton of direct motivation to support it.

I know how much it sucks to find a lib you'd like to use and it ends up having some stupid incompatibility or license issue that's *just* too annoying to deal with for you to end up using the lib. Accordingly, I do have some interest in making sure Cass is available and usable on many platforms/versions. Unfortunately, it comes down to the time investment for me to keep two separate branches up to date. As it stands, I'm already maintaining a Java version of the same library in addition to Cass, with a severe lack of consistent long-term help (nobody for Java and only Jason here).

I grant that a lot of future changes would be virtually identical, but the time investment to maintain these libs already cuts into my time for developing projects using them more than I'd like. If anything, I'd be creating an Android-specific branch of the Java version before a Python 2.x branch based on the interest I've seen.

That said, if you or someone else are interested in taking the lead on converting/maintaining a 2.x branch I'd be happy to help out.

TL;DR: My opinion on this lines up pretty closely with Jason's. I don't really have the time (or interest to make the time) to maintain a Python 2.x branch in addition to the 3.x version and the Java version right now. I'd be happy to help if someone with more interest wanted to take the lead on it though.
